ps-modules/Build/Build.psm1

```diff
@@ -0,0 +1,265 @@
+       isOnMacOS = $isOnMacOS
+       packageNameOnly = $packageNameOnly
+       osName = $osName
+       triplets = $triplets
+       preStagePath = $preStagePath
+       stagePath = $stagedArtifactsPath
+       artifactName = $artifactName
+       prebuildScript = "custom-steps/$packageNameOnly/pre-build.ps1"
+       postbuildScript = "custom-steps/$packageNameOnly/post-build.ps1"
+   }
+   if ($isOnMacOS) {
+       $vars.macArm64Dir = "$vcpkgInstallDir/installed/$($triplets[0])"
+       $vars.macX64Dir = "$vcpkgInstallDir/installed/$($triplets[1])"
+   }
+   Write-Debug "$(NL)Initialized vars:"
+   Write-Debug (Get-PSObjectAsFormattedList -Object $vars)
+   return $vars
+}
+
+function Setup-Vcpkg {
+   param(
+      [string]$repo,
+      [string]$repoHash,
+      [string]$installDir,
+      [string]$cacheDir,
+      [string]$bootstrapScript
+   )
+   Write-Banner -Level 3 -Title "Setting up vcpkg"
+   Write-Message "Removing vcpkg..."
+   Write-Debug "> Removing vcpkg system cache..."
+   Write-Debug ">> Looking for user-specific vcpkg cache dir: $cacheDir"
+   if (Test-Path -Path $cacheDir -PathType Container) {
+      Write-Debug ">> Deleting dir: $cacheDir"
+      Remove-Item -Path $cacheDir -Recurse -Force
+   } else {
+      Write-Debug ">> Directory not found: $cacheDir"
+   }    
+   Write-Debug "> Removing dir: vcpkg"
+   if (Test-Path -Path $installDir -PathType Container) {
+      Write-Debug ">> Directory found. Deleting: $installDir"
+      Remove-Item -Path $installDir -Recurse -Force
+   } else {
+      Write-Debug ">> Directory not found: $installDir, skipping step."
+   }
+
+   Write-Message "$(NL)Installing vcpkg..."
+   if (Test-Path -Path $installDir -PathType Container) {
+       Write-Debug "> Directory already exists: $installDir!!!"
+   }
+   else
+   {
+       Write-Debug "> Cloning repo: $repo"
+       git clone $repo
+       if ($repoHash -ne "") {
+           Write-Debug ">> Checking out hash: $repoHash"
+           Push-Location vcpkg
+           git checkout $repoHash
+           Pop-Location
+       }
+
+       Write-Debug "> Bootstrapping vcpkg..."
+       Push-Location $installDir
+       Invoke-Expression "$bootstrapScript"
+       Pop-Location
+   }
+}
+
+function Run-PreBuildScriptIfExists {
+   param(
+      $script
+   )
+   Run-ScriptIfExists -title "Pre-build step" -script $script
+}
+
+function Run-PostBuildScriptIfExists {
+   param(
+      $script,
+      $preStagePath
+   )
+   $scriptArgs = @{ BuildArtifactsPath = ((Resolve-Path $preStagePath).Path -replace '\\', '/') }
+   Run-ScriptIfExists -title "Post-build step" -script $script -scriptArgs $scriptArgs
+}
+
+function Install-Package
+{
+   param(
+      [string]$vcpkgExe,
+      [string]$package,
+      [PSObject]$triplets
+   )
+   Write-Banner -Level 3 -Title "Installing package: $package"
+   $tripletCount = $triplets.Length
+   Write-Message "Installing for $tripletCount triplet(s)..."
+
+   $tripletNum = 1
```

Comment:
Can we get rid of `tripletNum`? I thought maybe the `tripletNum` was part of a path on disk but it's just used in the `Write-Message`

Reply:
Yeah, I think we will only ever do 1 triplet for windows and 2 for mac, so perhaps it doesn't make sense to print out which triplet number you are on in the process.  I'll remove it.

Reply:
Removed as part of: 45c439a6e0fb6204884bd2f9584b08fdb7d576d0